bleak/backends/bumble/scanner.py

```diff
@@ -0,0 +1,125 @@
+    get_link,
+    start_transport,
+)
+from bleak.backends.scanner import (
+    AdvertisementData,
+    AdvertisementDataCallback,
+    BaseBleakScanner,
+)
+
+logger = logging.getLogger(__name__)
+
+SERVICE_UUID_TYPES = [
+    AdvertisingData.COMPLETE_LIST_OF_128_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.COMPLETE_LIST_OF_16_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.COMPLETE_LIST_OF_32_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_16_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_32_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_128_BIT_SERVICE_CLASS_UUIDS,
+]
+
+# Arbitrary BD_ADDR for the scanner device
+SCANNER_BD_ADDR = "F0:F1:F2:F3:F4:F5"
+
+
+class BleakScannerBumble(BaseBleakScanner, Device.Listener):
+    """
+    Interface for Bleak Bluetooth LE Scanners
+
+    Args:
+        detection_callback:
+            Optional function that will be called each time a device is
+            discovered or advertising data has changed.
+        service_uuids:
+            Optional list of service UUIDs to filter on. Only advertisements
+            containing this advertising data will be received.
+    Keyword Args:
+        adapter (BumbleTransport): Bumble transport adapter to use.
+    """
+
+    def __init__(
+        self,
+        detection_callback: Optional[AdvertisementDataCallback],
+        service_uuids: Optional[List[str]],
+        scanning_mode: Literal["active", "passive"],
+        **kwargs,
+    ):
+        super(BleakScannerBumble, self).__init__(detection_callback, service_uuids)
+        self.device = Device("scanner_dev", address=Address(SCANNER_BD_ADDR))
+
+        self.device.host = Host()
+        self.device.host.controller = Controller("Scanner", link=get_link())
+        self.device.listener = self
+        self._scan_active = scanning_mode == "active"
+        self._adapter: BumbleTransport = kwargs.get("adapter", get_default_transport())
+
+    def on_advertisement(self, advertisement: Advertisement):
+        service_uuids: List[str] = []
+        service_data: Dict[str, AdvertisingDataObject] = {}
+        local_name = advertisement.data.get(AdvertisingData.COMPLETE_LOCAL_NAME)
+        if not local_name:
+            local_name = advertisement.data.get(AdvertisingData.SHORTENED_LOCAL_NAME)
+        manuf_data = advertisement.data.get(AdvertisingData.MANUFACTURER_SPECIFIC_DATA)
+        for uuid_type in SERVICE_UUID_TYPES:
+            adv_uuids = advertisement.data.get(uuid_type)
+            if adv_uuids is None:
+                continue
+            if not isinstance(adv_uuids, list):
+                continue
+            for uuid in adv_uuids:  # type: UUID
+                if uuid not in service_uuids:
+                    service_uuids.append(str(uuid))
+
+        for service_data in advertisement.data.get_all(AdvertisingData.SERVICE_DATA):
+            service_uuid, data = service_data
+            service_data[str(service_uuid)] = data
```

Comment:
Incompatible types in assignment (expression has type "list[UUID] | tuple[UUID, bytes] | bytes | str | int | tuple[int, int] | tuple[int, bytes] | Appearance", variable has type "dict[str, list[UUID] | tuple[UUID, bytes] | bytes | str | int | tuple[int, int] | tuple[int, bytes] | Appearance]")

Reply:
Service data expects `dict[str, bytes`, but the for loop mutates the dict and fills some of the values with strings. Is this intentional?